File: exercise2/statically.js
class Statically {
  constructor(_glue, jello) {
    this._glue = _glue;
    this.jello = jello;
  }

  getGlue = function () {
    return "epoxy";
  };

  getJello = function () {
    return "jello";
  };
  moveAndShake() {
    console.log("Never stagnate in life.  Be agile, change, and improve...");
  }

  static stubborn(moveAndShake) {
    console.log("Please don't make me change... I just don't like it...");
  }

  static explain(stubborn) {
    console.log("epoxy");
    console.log(
      "Static keyword in Java is used to show you don't need to create an instance of the class. Static in javascript means the function is a utility type function that should not be instantiated but rather used as a template or just called to perform a function."
    );
  }
}

//get the below code to run correctly without errors:  Make sure you explain what you did in the the explain() method, defined in the class.  Do NOT remove the 'static' keyword from the explain() method.

const stats = new Statically();

stats.moveAndShake();
Statically.stubborn();
Statically.explain();

console.log("This is stats.getGlue: " + stats.getGlue);
console.log("This is stats.jello " + stats.getJello);

File: exercise2/exercise2/Statically.java
package exercise2;

public class Statically {
    private String glue = "Epoxy";
    public String jello = "Jello";

    public Statically(String glue, String jello) {
        this.glue = glue;
        this.jello = jello;
    }

    public String getGlue() {
        return glue;
    }

    public String getJello() {
        return jello;
    }

    public static int moveAndShake(int x, int y) {
        int moveShake = x * y;
        System.out.println("Never stagnate in life.  Be agile, change, and improve...");
        return moveShake;
    }

    public static int stubborn(int a, int b) {
        int stubs = a + b;
       Statically.moveAndShake(1, 2);
            System.out.println("Please don't make me change... I just don't like it...");
        return stubs;
    }


    public static int explain(int c, int d) {
        int expl = c + d;
        Statically.stubborn(1, 2);
        System.out.println("epoxy");
        System.out.println(
                "Static keyword in Java is used to show you don't need to create an instance of the class. Static in javascript means the function is a utility type function that should not be instantiated but rather used as a template or just called to perform a function."
        );
        return expl;
    }
}



